In [3]:
import pandas as pd
import numpy as np

df = pd.read_csv("../data/raw/Sambhajinagar_Dataset_V3_Final.csv")

print("Dataset loaded successfully")

Dataset loaded successfully


In [4]:
print("Dataset shape:", df.shape)

print("\nColumn names:")
print(df.columns.tolist())

print("\nData types:")
print(df.dtypes)

Dataset shape: (9893, 10)

Column names:
['system:index', 'Elevation', 'LST', 'LandCover', 'Latitude', 'Longitude', 'NDBI', 'NDVI', 'Population', '.geo']

Data types:
system:index      int64
Elevation         int64
LST             float64
LandCover         int64
Latitude        float64
Longitude       float64
NDBI            float64
NDVI            float64
Population      float64
.geo                str
dtype: object


In [5]:
print("Missing values per column:")
print(df.isnull().sum())

print("\nDuplicate rows:")
print(df.duplicated().sum())

print("\nFirst 5 rows:")
display(df.head())

Missing values per column:
system:index    0
Elevation       0
LST             0
LandCover       0
Latitude        0
Longitude       0
NDBI            0
NDVI            0
Population      0
.geo            0
dtype: int64

Duplicate rows:
0

First 5 rows:


,system:index,Elevation,LST,LandCover,Latitude,Longitude,NDBI,NDVI,Population,.geo
0,0,919,43.919170,40,19.970832,75.438700,0.127207,0.162286,1.562392,"{""type"":""Point"",""coordinates"":[75.438699969516..."
1,1,488,39.060455,40,19.729093,75.209548,-0.272423,0.570751,1.361367,"{""type"":""Point"",""coordinates"":[75.209547657446..."
2,2,505,40.231127,30,19.788775,75.225639,0.055877,0.099371,15.386780,"{""type"":""Point"",""coordinates"":[75.225638822974..."
3,3,757,44.884761,40,19.960297,75.544582,0.121404,0.196037,2.643492,"{""type"":""Point"",""coordinates"":[75.544582071461..."
4,4,586,48.502735,30,19.867709,75.486062,0.133151,0.165262,31.397209,"{""type"":""Point"",""coordinates"":[75.486061551192..."


In [6]:
landcover_counts = (
    df["LandCover"]
    .value_counts()
    .sort_index()
)

print("LandCover class counts:")
print(landcover_counts)

print("\nNumber of unique LandCover classes:", df["LandCover"].nunique())

LandCover class counts:
LandCover
10     483
20     714
30    2421
40    5253
50     839
60     154
80      29
Name: count, dtype: int64

Number of unique LandCover classes: 7


In [7]:
numeric_summary = df[
    [
        "Elevation",
        "LST",
        "Latitude",
        "Longitude",
        "NDBI",
        "NDVI",
        "Population"
    ]
].describe().T

numeric_summary

,count,mean,std,min,25%,50%,75%,max
Elevation,9893.0,621.370262,85.589970,483.000000,556.000000,597.000000,691.000000,926.000000
LST,9893.0,43.463630,3.229270,27.762190,41.241152,43.606422,45.966564,51.956644
Latitude,9893.0,19.890409,0.085958,19.723374,19.819395,19.892116,19.963205,20.042864
Longitude,9893.0,75.357082,0.118384,75.142142,75.255050,75.356124,75.454214,75.586524
NDBI,9893.0,0.029918,0.098665,-0.433368,-0.021819,0.053535,0.099973,0.401310
NDVI,9893.0,0.241080,0.115880,-0.128937,0.160441,0.211398,0.296107,0.751738
Population,9893.0,12.378498,29.300305,0.667908,2.063318,3.155722,7.110859,230.774673


In [9]:
audit_summary = pd.DataFrame({
    "Metric": [
        "Rows",
        "Columns",
        "Missing Values",
        "Duplicate Rows",
        "Unique LandCover Classes"
    ],
    "Value": [
        df.shape[0],
        df.shape[1],
        int(df.isnull().sum().sum()),
        int(df.duplicated().sum()),
        df["LandCover"].nunique()
    ]
})

audit_summary

,Metric,Value
0,Rows,9893
1,Columns,10
2,Missing Values,0
3,Duplicate Rows,0
4,Unique LandCover Classes,7


In [10]:
import os

os.makedirs("../outputs/reports", exist_ok=True)

audit_summary.to_csv(
    "../outputs/reports/data_audit_summary_v3.csv",
    index=False
)

print("Saved: outputs/reports/data_audit_summary_v3.csv")

Saved: outputs/reports/data_audit_summary_v3.csv


## Dataset V3 Audit Conclusion

Dataset V3 contains 9,893 sampled locations and 10 original columns. No missing values or duplicate rows were found.

The dataset includes seven ESA WorldCover land-cover classes: Tree cover, Shrubland, Grassland, Cropland, Built-up land, Bare/sparse vegetation, and Permanent water bodies.

The `LandCover` field is categorical and will be one-hot encoded during feature engineering. The `system:index` and `.geo` fields will be removed during data cleaning. Latitude and Longitude will be retained only for mapping and spatial block cross-validation, not for model training.